src/engine/TransitivePathBase.cpp

```diff
@@ -301,7 +301,13 @@ VariableToColumnMap TransitivePathBase::computeVariableToColumnMap() const {
 
 // _____________________________________________________________________________
 bool TransitivePathBase::knownEmptyResult() {
-  return subtree_->knownEmptyResult();
+  // When minDist_ == 0, then there always needs to be a starting tree bound for
+  // correct results, so it is safe to call `std::optional<>::value()` without a
+  // prior check.
+  return subtree_->knownEmptyResult() &&
+         (minDist_ > 0 || decideDirection()
+                              .first.treeAndCol_.value()
+                              .first->knownEmptyResult());
```

Comment:
```suggestion
  return (subtree_->knownEmptyResult() &&
         minDist_ > 0) || (decideDirection()
                              .first.treeAndCol_.value()
                              .first->knownEmptyResult());
```

Isn't that also correct and covers more cases?

so ` ?x <nonExistingIri>/<ExistingIri>* ?y` is also known to be empty.
In that case you probably can't do the `.value()` unchecked, because it might be unbound (in which case you can't say anything and have to return `false`).

Reply:
Yes, that is absolutely correct (except for the missing has_value check that is).